src/SprykerEco/Zed/Econda/Business/EcondaBusinessFactory.php

```diff
@@ -0,0 +1,285 @@
+     */
+    public function createFileCategoryCollector()
+    {
+        $storageCategoryNodeCollector = new EcondaCategoryCollector(
+            $this->createCriteriaBuilder(),
+            $this->createStoragePdoQueryAdapterByName('CategoryNodeEcondaQuery')
+        );
+
+        return $storageCategoryNodeCollector;
+    }
+
+    /**
+     * @return \SprykerEco\Zed\Econda\Business\Collector\File\EcondaProductCollector
+     */
+    public function createFileProductCollector()
+    {
+        /**
+         * @var \SprykerEco\Zed\Econda\Business\Collector\File\EcondaProductCollector $storageProductCollector
+         */
+        $storageProductCollector = new EcondaProductCollector(
+            $this->createCriteriaBuilder(),
+            $this->createStoragePdoQueryAdapterByName('ProductConcreteEcondaQuery'),
+            $this->getProductCategoryQueryContainer(),
+            $this->getProductImageQueryContainer(),
+            $this->getPriceFacade(),
+            $this->getConfig()
+        );
+
+        return $storageProductCollector;
+    }
+
+    /**
+     * @return \SprykerEco\Zed\Econda\Business\Exporter\ExporterInterface
+     */
+    protected function createFileExporter()
+    {
+        $fileExporter = new FileExporter(
+            $this->createFileWriter(),
+            $this->createFailedResultModel(),
+            $this->createBatchResultModel(),
+            $this->createCsvNameGenerator(),
+            $this->getConfig()->getFileExportPath(),
+            $this->getCollectorFileExporterPlugins()
+        );
+
+        return $fileExporter;
+    }
+
+    /**
+     * @return \SprykerEco\Zed\Econda\Dependency\Plugin\EcondaPluginInterface[]
+     */
+    protected function getCollectorFileExporterPlugins()
+    {
+        return $this->getProvidedDependency(EcondaDependencyProvider::FILE_PLUGINS);
+    }
+
+    /**
+     * @return \SprykerEco\Zed\Econda\Dependency\Facade\EcondaToLocaleInterface
+     */
+    protected function getLocaleFacade()
+    {
+        return $this->getProvidedDependency(EcondaDependencyProvider::FACADE_LOCALE);
+    }
+
+    /**
+     * @return \SprykerEco\Zed\Econda\Persistence\EcondaQueryContainer
+     */
+    protected function getEcondaQueryContainer()
+    {
+        return new EcondaQueryContainer();
+    }
+
+    /**
+     * @return \SprykerEco\Zed\Econda\Business\Exporter\Writer\File\Adapter\CsvAdapter
+     */
+    protected function createCsvFileWriterAdapter()
+    {
+        return new CsvAdapter($this->getConfig()->getFileExportPath());
+    }
+
+    /**
+     * @return \Generated\Shared\Transfer\FailedResultTransfer
+     */
+    protected function createFailedResultModel()
+    {
+        return new FailedResultTransfer();
+    }
+
+    /**
+     * @return string
+     */
+    protected function getCurrentDatabaseEngineName()
+    {
+        return $this->getPropelFacade()->getCurrentDatabaseEngineName();
```

Comment:
Here the factory is also abused somewhat. I think you should only get the external facade and then call the method you want to use directly from business logic.

Reply:
The thing here is that:
- getCurrentDatabaseEngineName is used in createStoragePdoQueryAdapterByName
- which in turn is used in createEcondaCategoryCollector
and all those methods are inside the EcondaBusinessFactory.
The EcondaFacade just uses the createEcondaCategoryCollector to get the collector